host/taskpoller.go

```diff
@@ -454,7 +454,8 @@ func (p *TaskPoller) PollAndProcessActivityTaskWithID(dropTask bool) error {
 }
 
 func createContext() context.Context {
-	ctx, _ := context.WithTimeout(context.Background(), 90*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), 90*time.Second)
+        defer cancel() 
```

Comment:
the context will be cancelled immediately after this function returns. We should also return cancel and let caller to call cancel.
BTW, this code is only used in our integration test. It's not good but it's ok.

Reply:
If you run "go vet" over your code, it should point out mistakes of this kind. For example:
https://go.dev/play/p/1zlF7qFbzhY

Reply:
hey @adonovan, the proper fix would be to return cancel callback and change all the callers with `defer cancel()`. Do you want to proceed with that? Otherwise let's close this PR and we will find someone else to address it

Reply:
I have prepared an updated patch. However, I am traveling now and cannot access the GitHub account used to create the pull request. I will address this in about a week.